AEPCore/Sources/eventhub/EventHub.swift

```diff
@@ -140,14 +140,14 @@ final class EventHub {
     ///   - timeout A timeout in seconds, if the response listener is not invoked within the timeout, then the `EventHub` invokes the response listener with a nil `Event`
     ///   - listener: An `EventResponseListener` which will be invoked whenever the `EventHub` receives the response `Event` for `triggerEvent`
     func registerResponseListener(triggerEvent: Event, timeout: TimeInterval, listener: @escaping EventResponseListener) {
-        var responseListenerContainer: EventListenerContainer? // initialized here so we can use in timeout block
-        responseListenerContainer = EventListenerContainer(listener: listener, triggerEventId: triggerEvent.id, timeout: DispatchWorkItem { [weak self] in
+        let triggerEventId = triggerEvent.id
+        let responseListenerContainer = EventListenerContainer(listener: listener, triggerEventId: triggerEventId, timeout: DispatchWorkItem { [weak self, triggerEventId] in
             listener(nil)
-            _ = self?.responseEventListeners.filterRemove { $0 == responseListenerContainer }
+            _ = self?.responseEventListeners.filterRemove { $0.triggerEventId == triggerEventId }
         })
 
-        DispatchQueue.global().asyncAfter(deadline: DispatchTime.now() + timeout, execute: responseListenerContainer!.timeoutTask!)
-        responseEventListeners.append(responseListenerContainer!)
+        DispatchQueue.global().asyncAfter(deadline: DispatchTime.now() + timeout, execute: responseListenerContainer.timeoutTask!)
```

Comment:
is the `responseListenerContainer` guaranteed to have a `timeoutTask`?  i know this isn't your code, but we've tried to avoid force unwrapping as much as possible.

Reply:
In this case, yes but i have refactored the code to avoid force unwrapping.